cmd/root.go

```diff
@@ -77,6 +78,17 @@ func runRootCmd(cmd *cobra.Command, args []string) {
 			if err = viper.ReadInConfig(); err != nil {
 				logger.Fatalw("failed to read config", zap.Error(err))
 			}
+
+			var inputs []interface{}
+			for _, v := range inputsource.All() {
+				inputs = append(inputs, v)
+			}
+			viper.Set("input_sources.primary", inputs)
+
+			err = viper.WriteConfig()
+			if err != nil {
```

Comment:
I believe this could be simplified to `if err = viper.WriteConfig(); err != nil { ... }`